### ISSUE TITLE
Gradle cache restoration issue

### ISSUE THREAD

**Issue by plastiv** [arturbosch/detekt#2180]
## Expected Behavior

Running gradle detekt task from gradle plugin on the same kotlin sources should reuse gradle task cache instead of executing task twice. 
## Observed Behavior

Missing html output file prevents cache entity being reused when running detekt task with gradle plugin. 

I believe this is due to missing ` @get:PathSensitive(PathSensitivity.RELATIVE)` annotation on report output files: https://github.com/arturbosch/detekt/blob/5f41ea56e1915b86271bfefdcc7b68e7650e6cab/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/Detekt.kt#L145-L158

## Steps to Reproduce

* have a project with detekt plugin integrated, like `my-project1`
* make sure for a project local cache is enabled, remote cache is disabled
* copy project into different folder on disk, like `my-project2` to trigger different PATH issue
* remove local cache to avoid accidental artifact reuse, like `rm -rf ~/.gradle/caches`
* run `./gradlew clean detekt` at `my-project1` to seed a cache entry
* run `./gradlew clean detekt` at `my-project2`. 

Expected: detekt task `FROM-CACHE`. Actual: detekt task is executed due to `Missing html output file` property.

## Context

Detekt tasks are marked as cooperative and are participating in cache lifecycle by uploading and downloading task artifacts. yet, their cache can't be reused when running on CI with ephemeral nodes due to jobs running project at different folder paths. 

## Your Environment

* Version of detekt used: 1.2.2
* Version of Gradle used (if applicable): 6.0.1
* Operating System and version: mac and linux
* Link to your project (if it's a public repository): if you are not sure how to reproduce the issue, let me know I will try to find some time to work on a sample

**Comment by plastiv** [comment]
I have managed to create minimal reproducible example. Run `./test.sh` to trigger steps-to-reproduce from above. 

* https://github.com/plastiv/detekt-cache-issue
* https://travis-ci.org/plastiv/detekt-cache-issue/builds/625774155

I have spent more time investigating the issue and currently believe that it isn't the html report output file is a culprit here. 

Instead, it's a source input folder. `Detekt` task extends `SourceTask` which according to [documentation](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.SourceTask.html#N1D5ED) has `source` property marked as `PathSensitivity.ABSOLUTE` having task inputs with absolute path sensitivity effectively disables cache reusability: https://docs.gradle.org/current/userguide/build_cache.html#sec:task_output_caching_inputs
